transport/graphsync/graphsync.go

```diff
@@ -21,6 +22,13 @@ var log = logging.Logger("dt_graphsync")
 
 var errContextCancelled = errors.New("context cancelled")
 
+// When restarting a data transfer, we cancel the existing graphsync request
+// before opening a new one.
+// These constants define the minimum and maximum time to wait for the request
+// to be cancelled.
+const minGSCancelWait = 100 * time.Millisecond
+const maxGSCancelWait = time.Second
```

Comment:
Overall LGTM, I am just not sure how we decide that these values are reasonable...

Reply:
It's true, ideally there would be a way to ask graphsync to cancel a request and wait until all events have been emitted before returning